internal/cfgplugins/interface.go

```diff
@@ -927,3 +927,120 @@ func NewAggregateInterface(t *testing.T, dut *ondatra.DUTDevice, b *gnmi.SetBatc
 	}
 	return agg
 }
+
+// StaticARPEntry defines per-port static ARP mapping.
+type StaticARPEntry struct {
+	PortName string // DUT port name (e.g., "port2")
+	MagicIP  string // Per-port IP (e.g., "192.0.2.1")
+	MagicMAC string // Per-port MAC (e.g., "00:1A:2B:3C:4D:5E")
+}
+
+// StaticARPConfig holds all per-port static ARP entries.
+type StaticARPConfig struct {
+	Entries []StaticARPEntry
+}
+
+// StaticARPWithMagicUniversalIP configures static ARP and static routes per-port.
+func StaticARPWithMagicUniversalIP(t *testing.T, dut *ondatra.DUTDevice, cfg StaticARPConfig) {
+	t.Helper()
+	sb := &gnmi.SetBatch{}
```

Comment:
the batch object should be a parameter. See [cfgplugins README](https://github.com/openconfig/featureprofiles/blob/main/internal/cfgplugins/README.md).